skins/oasis/modules/PageHeaderController.class.php

```diff
@@ -53,9 +50,9 @@ protected function prepareActionButton() {
 
 		// handle protected pages (they should have viewsource link and lock icon) - BugId:9494
 		if ( isset( $this->content_actions['viewsource'] ) &&
-			!$wgTitle->isProtected() &&
-			!$wgTitle->isNamespaceProtected( $wgUser ) &&
-			!$wgUser->isLoggedIn() /* VOLDEV-74: logged in users should see the viewsource button, not edit */
+			!$this->wg->Title->isProtected() &&
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)